tests/functional/settings_common.js

```diff
@@ -0,0 +1,207 @@
+    name: 'visiting settings pages unverified',
+
+    beforeEach: function () {
+      email = TestHelpers.createEmail();
+
+      client = new FxaClient(AUTH_SERVER_ROOT, {
+        xhr: nodeXMLHttpRequest.XMLHttpRequest
+      });
+
+      var self = this;
+      return client.signUp(email, FIRST_PASSWORD)
+              .then(function () {
+                return FunctionalHelpers.clearBrowserState(self);
+              });
+    },
+
+    teardown: function () {
+      return FunctionalHelpers.clearBrowserState(this);
+    }
+  };
+
+  function unverifiedAccountTest (suite, page) {
+    suite['visit settings' + page + ' with an unverified account redirects to confirm'] = function () {
+      var self = this;
+      var url = SETTINGS_URL + page;
+
+      return self.get('remote')
+        .get(require.toUrl(SIGNIN_URL))
+        .setFindTimeout(intern.config.pageLoadTimeout)
+        .findByCssSelector('form input.email')
+          .click()
+          .type(email)
+        .end()
+
+        .findByCssSelector('form input.password')
+          .click()
+          .type(FIRST_PASSWORD)
+        .end()
+
+        .findByCssSelector('button[type="submit"]')
+          .click()
+        .end()
+
+        .findById('fxa-confirm-header')
+        .end()
+
+        .get(require.toUrl(url))
+        // Expect to get redirected to confirm since the account is unverified
+        .findById('fxa-confirm-header')
+        .end();
+    };
+  }
+
+  var verifiedSuite = {
+    name: 'settings common',
+
+    beforeEach: function () {
+      email = TestHelpers.createEmail();
+
+      client = new FxaClient(AUTH_SERVER_ROOT, {
+        xhr: nodeXMLHttpRequest.XMLHttpRequest
+      });
+
+      var self = this;
+      return client.signUp(email, FIRST_PASSWORD, { preVerified: true })
+              .then(function (result) {
+                accountData = result;
+                return FunctionalHelpers.clearBrowserState(self);
+              });
+    },
+
+    teardown: function () {
+      return FunctionalHelpers.clearBrowserState(this);
+    }
+  };
+
+  function verifiedAccountTest (suite, page, pageHeader) {
+    var url = SETTINGS_URL + page;
+
+    suite['visit settings' + page + ' with an invalid sessionToken redirects to signin'] = function () {
+      // Changing the password invalidates the current sessionToken
+      var self = this;
+
+      return FunctionalHelpers.clearBrowserState(self)
+        .then(function () {
+          return client.passwordChange(email, FIRST_PASSWORD, SECOND_PASSWORD);
+        })
+        .then(function () {
+          return self.get('remote')
+            .get(require.toUrl(url))
+            // Expect to get redirected to sign in since the sessionToken is invalid
+            .findById('fxa-signin-header')
+            .end();
+        });
+    };
+
+    suite['visit settings' + page + ' with an unknown uid parameter redirects to signin'] = function () {
+      var self = this;
+
+      return self.get('remote')
```

Comment:
Fixed– but we need to set load the SIGNIN_URL here to make sure it doesn't have any query params.